### PR TITLE
dev-python/logbook: version bump to 1.5.3

### PR DESCRIPTION
**Pull request by ZeroPointEnergy**
Closes: https://bugs.gentoo.org/719082
Package-Manager: Portage-2.3.89, Repoman-2.3.20
Signed-off-by: Andreas Zuber <a.zuber@gmx.ch>

Tests Results with python 3.6, 3.7, 3.8
<details>
>>> Test phase: dev-python/logbook-1.5.3
 * python3_6: running distutils-r1_run_phase python_test
======================== test session starts =========================
platform linux -- Python 3.6.10, pytest-5.4.1, py-1.8.0, pluggy-0.13.1
rootdir: /var/tmp/portage/dev-python/logbook-1.5.3/work/logbook-1.5.3-python3_6
collected 199 items

tests/test_asyncio.py s                                        [  0%]
tests/test_ci.py ss                                            [  1%]
tests/test_deadlock.py .                                       [  2%]
tests/test_file_handler.py ................................    [ 18%]
tests/test_fingers_crossed_handler.py ......                   [ 21%]
tests/test_flags.py ..                                         [ 22%]
tests/test_groups.py ..                                        [ 23%]
tests/test_handler_errors.py ...                               [ 24%]
tests/test_handlers.py ...........                             [ 30%]
tests/test_helpers.py ..                                       [ 31%]
tests/test_log_record.py ..........                            [ 36%]
tests/test_logbook.py ...                                      [ 37%]
tests/test_logger.py ...                                       [ 39%]
tests/test_logging_api.py .............                        [ 45%]
tests/test_logging_compat.py ....                              [ 47%]
tests/test_logging_times.py .....                              [ 50%]
tests/test_mail_handler.py .......                             [ 53%]
tests/test_more.py .........sss                                [ 59%]
tests/test_nteventlog_handler.py s                             [ 60%]
tests/test_null_handler.py ......                              [ 63%]
tests/test_processors.py ....                                  [ 65%]
tests/test_syslog_handler.py ............                      [ 71%]
tests/test_test_handler.py ..............................      [ 86%]
tests/test_ticketing.py .                                      [ 86%]
tests/test_unicode.py .......                                  [ 90%]
tests/test_utils.py ...................                        [100%]

========================== warnings summary ==========================
tests/test_ticketing.py::test_basic_ticketing
  /usr/lib64/python3.6/site-packages/sqlalchemy/dialects/sqlite/base.py:1466: SADeprecationWarning: The create_engine.convert_unicode parameter and corresponding dialect-level parameters are deprecated, and will be removed in a future release.  Modern DBAPIs support Python Unicode natively and this parameter is unnecessary.
    default.DefaultDialect.__init__(self, **kwargs)

tests/test_ticketing.py::test_basic_ticketing
  /var/tmp/portage/dev-python/logbook-1.5.3/work/logbook-1.5.3-python3_6/logbook/ticketing.py:246: SADeprecationWarning: The FromClause.count() method is deprecated, and will be removed in a future release.   Please use the functions.count function available from the func namespace.
    return self.engine.execute(self.tickets.count()).fetchone()[0]

-- Docs: https://docs.pytest.org/en/latest/warnings.html
============= 192 passed, 7 skipped, 2 warnings in 3.93s =============
 * python3_7: running distutils-r1_run_phase python_test
======================== test session starts =========================
platform linux -- Python 3.7.7, pytest-5.4.1, py-1.8.0, pluggy-0.13.1
rootdir: /var/tmp/portage/dev-python/logbook-1.5.3/work/logbook-1.5.3-python3_7
collected 199 items

tests/test_asyncio.py .                                        [  0%]
tests/test_ci.py ss                                            [  1%]
tests/test_deadlock.py .                                       [  2%]
tests/test_file_handler.py ................................    [ 18%]
tests/test_fingers_crossed_handler.py ......                   [ 21%]
tests/test_flags.py ..                                         [ 22%]
tests/test_groups.py ..                                        [ 23%]
tests/test_handler_errors.py ...                               [ 24%]
tests/test_handlers.py ...........                             [ 30%]
tests/test_helpers.py ..                                       [ 31%]
tests/test_log_record.py ..........                            [ 36%]
tests/test_logbook.py ...                                      [ 37%]
tests/test_logger.py ...                                       [ 39%]
tests/test_logging_api.py .............                        [ 45%]
tests/test_logging_compat.py ....                              [ 47%]
tests/test_logging_times.py .....                              [ 50%]
tests/test_mail_handler.py .......                             [ 53%]
tests/test_more.py .........sss                                [ 59%]
tests/test_nteventlog_handler.py s                             [ 60%]
tests/test_null_handler.py ......                              [ 63%]
tests/test_processors.py ....                                  [ 65%]
tests/test_syslog_handler.py ............                      [ 71%]
tests/test_test_handler.py ..............................      [ 86%]
tests/test_ticketing.py .                                      [ 86%]
tests/test_unicode.py .......                                  [ 90%]
tests/test_utils.py ...................                        [100%]

========================== warnings summary ==========================
tests/test_ticketing.py::test_basic_ticketing
  /usr/lib/python3.7/site-packages/sqlalchemy/dialects/sqlite/base.py:1466: SADeprecationWarning: The create_engine.convert_unicode parameter and corresponding dialect-level parameters are deprecated, and will be removed in a future release.  Modern DBAPIs support Python Unicode natively and this parameter is unnecessary.
    default.DefaultDialect.__init__(self, **kwargs)

tests/test_ticketing.py::test_basic_ticketing
  /var/tmp/portage/dev-python/logbook-1.5.3/work/logbook-1.5.3-python3_7/logbook/ticketing.py:246: SADeprecationWarning: The FromClause.count() method is deprecated, and will be removed in a future release.   Please use the functions.count function available from the func namespace.
    return self.engine.execute(self.tickets.count()).fetchone()[0]

-- Docs: https://docs.pytest.org/en/latest/warnings.html
============= 193 passed, 6 skipped, 2 warnings in 4.04s =============
 * python3_8: running distutils-r1_run_phase python_test
======================== test session starts =========================
platform linux -- Python 3.8.2, pytest-5.4.1, py-1.8.0, pluggy-0.13.1
rootdir: /var/tmp/portage/dev-python/logbook-1.5.3/work/logbook-1.5.3-python3_8
collected 199 items

tests/test_asyncio.py .                                        [  0%]
tests/test_ci.py ss                                            [  1%]
tests/test_deadlock.py .                                       [  2%]
tests/test_file_handler.py ................................    [ 18%]
tests/test_fingers_crossed_handler.py ......                   [ 21%]
tests/test_flags.py ..                                         [ 22%]
tests/test_groups.py ..                                        [ 23%]
tests/test_handler_errors.py ...                               [ 24%]
tests/test_handlers.py ...........                             [ 30%]
tests/test_helpers.py ..                                       [ 31%]
tests/test_log_record.py ..........                            [ 36%]
tests/test_logbook.py ...                                      [ 37%]
tests/test_logger.py ...                                       [ 39%]
tests/test_logging_api.py .............                        [ 45%]
tests/test_logging_compat.py ....                              [ 47%]
tests/test_logging_times.py .....                              [ 50%]
tests/test_mail_handler.py .......                             [ 53%]
tests/test_more.py s........sss                                [ 59%]
tests/test_nteventlog_handler.py s                             [ 60%]
tests/test_null_handler.py ......                              [ 63%]
tests/test_processors.py ....                                  [ 65%]
tests/test_syslog_handler.py ............                      [ 71%]
tests/test_test_handler.py ..............................      [ 86%]
tests/test_ticketing.py s                                      [ 86%]
tests/test_unicode.py .......                                  [ 90%]
tests/test_utils.py ...................                        [100%]

=================== 191 passed, 8 skipped in 1.75s ===================
</details>

### CODE REVIEW

**Comment by ZeroPointEnergy** [comment]
Also pinging @juippis since he asked to be notified in #15254

File: dev-python/logbook/logbook-1.5.3.ebuild [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+PATCHES=( "${FILESDIR}"/${PN}-0.4.2-objectsinv.patch )
+
+python_prepare_all() {
+	# Delete test file requiring local connection to redis server
+	rm tests/test_queues.py || die
+	distutils-r1_python_prepare_all
+}
+
+python_compile_all() {
```
Comment (mgorny) [inline dev-python/logbook/logbook-1.5.3.ebuild:40]: This and below are redundant now.

File: dev-python/logbook/logbook-1.5.3.ebuild [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+distutils_enable_tests pytest
+distutils_enable_sphinx docs
+
+DEPEND="
+	dev-python/setuptools[${PYTHON_USEDEP}]
+	test? ( app-arch/brotli[${PYTHON_USEDEP},python] )
+	doc? ( >=dev-python/sphinx-1.1.3-r3[${PYTHON_USEDEP}] )"
+RDEPEND="dev-python/redis-py[${PYTHON_USEDEP}]"
+
+PATCHES=( "${FILESDIR}"/${PN}-0.4.2-objectsinv.patch )
```
Comment (mgorny) [inline dev-python/logbook/logbook-1.5.3.ebuild:32]: Do you still need this patch? If it's for intersphinx, d_e_s should handle it.

Comment (ZeroPointEnergy) [reply dev-python/logbook/logbook-1.5.3.ebuild]: Removed the patch, it works without it

File: dev-python/logbook/logbook-1.5.3.ebuild (hunk 1 above)
Comment (ZeroPointEnergy) [reply dev-python/logbook/logbook-1.5.3.ebuild]: ok

File: dev-python/logbook/logbook-1.5.3.ebuild [hunk 3]
```diff
@@ -0,0 +1,36 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+PYTHON_COMPAT=( python3_{6,7,8} )
+DISTUTILS_IN_SOURCE_BUILD=1
```
Comment (mgorny) [inline dev-python/logbook/logbook-1.5.3.ebuild:7]: Did you verify whether this is necessary?

File: dev-python/logbook/logbook-1.5.3.ebuild [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+DISTUTILS_USE_SETUPTOOLS=bdepend
```
Comment (mgorny) [inline dev-python/logbook/logbook-1.5.3.ebuild:8]: Redundant.

File: dev-python/logbook/logbook-1.5.3.ebuild [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+KEYWORDS="~amd64 ~x86"
+IUSE=""
+
+distutils_enable_tests pytest
+distutils_enable_sphinx docs
+
+DEPEND="
+	dev-python/setuptools[${PYTHON_USEDEP}]
+	test? ( app-arch/brotli[${PYTHON_USEDEP},python] )
+	doc? ( >=dev-python/sphinx-1.1.3-r3[${PYTHON_USEDEP}] )"
```
Comment (mgorny) [inline dev-python/logbook/logbook-1.5.3.ebuild:29]: Redundant.

File: dev-python/logbook/logbook-1.5.3.ebuild [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+LICENSE="BSD"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE=""
+
+distutils_enable_tests pytest
+distutils_enable_sphinx docs
+
+DEPEND="
+	dev-python/setuptools[${PYTHON_USEDEP}]
```
Comment (mgorny) [inline dev-python/logbook/logbook-1.5.3.ebuild:27]: Redundant.

File: dev-python/logbook/logbook-1.5.3.ebuild [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+
+LICENSE="BSD"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE=""
+
+distutils_enable_tests pytest
+distutils_enable_sphinx docs
+
+DEPEND="
```
Comment (mgorny) [inline dev-python/logbook/logbook-1.5.3.ebuild:26]: BDEPEND in EAPI 7.

File: dev-python/logbook/logbook-1.5.3.ebuild (hunk 3 above)
Comment (ZeroPointEnergy) [reply dev-python/logbook/logbook-1.5.3.ebuild]: It builds without it and the tests are passing

File: dev-python/logbook/logbook-1.5.3.ebuild (hunk 4 above)
Comment (ZeroPointEnergy) [reply dev-python/logbook/logbook-1.5.3.ebuild]: removed

File: dev-python/logbook/logbook-1.5.3.ebuild (hunk 5 above)
Comment (ZeroPointEnergy) [reply dev-python/logbook/logbook-1.5.3.ebuild]: removed

File: dev-python/logbook/logbook-1.5.3.ebuild (hunk 6 above)
Comment (ZeroPointEnergy) [reply dev-python/logbook/logbook-1.5.3.ebuild]: removed

File: dev-python/logbook/logbook-1.5.3.ebuild (hunk 7 above)
Comment (ZeroPointEnergy) [reply dev-python/logbook/logbook-1.5.3.ebuild]: changed

**Comment by mgorny** [comment]
1. Speedups fail to build for some reason.
2. A lot of tests fail, then `tests/test_syslog_handler.py::test_syslog_handler[False-ContextEnteringStrategy-None-AddressFamily.AF_INET-SocketKind.SOCK_STREAM-address1]` plain hangs.

I'll try to skip that test to get proper failure output.

[dev-python:logbook-1.5.3:20200424-141647.log](https://github.com/gentoo/gentoo/files/4529439/dev-python.logbook-1.5.3.20200424-141647.log)

**Comment by mgorny** [comment]
Here are more verbose failures:
[dev-python:logbook-1.5.3:20200424-142233.log](https://github.com/gentoo/gentoo/files/4529459/dev-python.logbook-1.5.3.20200424-142233.log)

**Comment by ZeroPointEnergy** [comment]
Hmm, the test all pass on my system for all three python versions. Will take a look at the output

**Comment by ZeroPointEnergy** [comment]
The speedup error comes from a missing step I found in the travis file. It builds some C extensions with cython. I made this optional and set the environment variable to actively disable it when the use flag is not set, which gets rid of the error message.

I created a separate commit for this to hopefully make the review easier. I will squash them again in the end.

I'm still looking into the test errors. I guess there are some dependencies missing for the tests still. Is there an easy way to get a clean test environment so I can reproduce the errors? What are you people using for this?

**Comment by ZeroPointEnergy** [comment]
I get the following error when using speedup with pyfa:
```
python3.7: logbook/_speedups.c:6835: __pyx_f_7logbook_9_speedups_19ContextStackManager_iter_context_objects: Assertion `PyList_Check(__pyx_t_1)' failed.
/var/tmp/portage/games-util/pyfa-2.20.2-r1/temp/environment: line 2650:   131 Aborted                 python ./db_update.py
```
something is not right.

**Comment by mgorny** [comment]
Heh, I thought speedups fail for me only and that is the cause for test failures. If they didn't work for you, it's gonna be tough to debug. If you have some time, you could try installing the pytest plugins I have and seeing if that changes anything:

```
[I] dev-python/pytest-aiohttp (0.3.0{xpak}@17.03.2020): pytest plugin for aiohttp support
[I] dev-python/pytest-asyncio (0.11.0{xpak}@21.04.2020): Library for testing asyncio code with pytest
[U] dev-python/pytest-django (3.8.0@08.03.2020 -> (~)3.9.0^t{xpak}): A Django plugin for py.test
[I] dev-python/pytest-expect (1.1.0-r1@03.03.2020): py.test plugin that stores test expectations by saving the set of failing tests
[U] dev-python/pytest-fixture-config (1.7.0{xpak}@23.03.2020 -> (~)1.7.0-r1^t): Virtualenv fixture for py.test
[I] dev-python/pytest-httpbin (1.0.0{xpak}@19.03.2020): Easily test your HTTP library against a local copy of httpbin
[I] dev-python/pytest-mock (3.1.0{xpak}@19.04.2020): Thin-wrapper around the mock package for easier use with pytest
[I] dev-python/pytest-shutil (1.7.0-r1{xpak}@04.04.2020): A goodie-bag of unix shell and environment tools for py.test
[I] dev-python/pytest-virtualenv (1.7.0-r1{xpak}@04.04.2020): Virtualenv fixture for py.test
[I] dev-python/pytest-xprocess (0.13.1{xpak:3}@16.04.2020): Manage external processes across test runs
```

As for speedup crash, I suspect upstream is using some crappy Ubuntu or alike that builds everything with `-DNDEBUG` and they never run assertions.

**Comment by ZeroPointEnergy** [comment]
Ok, I should have done this a lot earlier. I read trough the setup.py and there are definitely some more dependencies. When I look at their documentation they seem to be optional. This is confirmed by the fact that if I install it with pip in a virtualenv it  does not install additional dependencies.

I listed all the dependencies as test dependencies as suggested by the python ebuild documentation. I don't think we need to make them optional as useflags because an application who uses such an optional extension will have to specify the dependency on it's own anyway.

Further I removed the speedup stuff for now (cython). This was never active in Gentoo anyway so far. I explicitly disabled it in the configure step so there is no error during the build anymore. The test failures have nothing to do with this, but when active I found at least one application who used logbook crashing. Should I open a ticket for this so we can inspect it further or at least track the issue? I don't think this optional feature that was never used in Gentoo so far should block the fixing of the ebuild and the python 3.7 rollout.

As for the tests, they hopefully run now on a clean system as well. I accidentally had all the libs installed on my system from other dependencies, this is why they always passed for me. I currently look into options for a clean test setup to verify that.

**Comment by mgorny** [comment]
Nope, same result still. I had all the deps installed already.

File: dev-python/logbook/logbook-1.5.3.ebuild [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+distutils_enable_tests pytest
+distutils_enable_sphinx docs
+
+BDEPEND="
+	test? (
+		app-arch/brotli[${PYTHON_USEDEP},python]
+		dev-python/execnet[${PYTHON_USEDEP}]
+		dev-python/jinja[${PYTHON_USEDEP}]
+		dev-python/pyzmq[${PYTHON_USEDEP}]
+		dev-python/redis-py[${PYTHON_USEDEP}]
```
Comment (mgorny) [inline dev-python/logbook/logbook-1.5.3.ebuild:30]: Is this actually used during the tests, given that you remove the test below?

**Comment by mgorny** [comment]
Good news and bad news: I've been able to get the tests to pass on a clean chroot. But this means we've got a few hundred packages that might break this.

**Comment by mgorny** [comment]
Ok, crazy idea: could you mail me your `find /var/db/pkg -mindepth 2 -maxdepth 2`? I'll diff it against my system, and see which packages I have in addition to yours.

**Comment by ZeroPointEnergy** [comment]
I can now reproduce at least two of the failing tests when I install `dev-python/brotlipy`.

Will send you the list.

File: dev-python/logbook/logbook-1.5.3.ebuild (hunk 8 above)
Comment (ZeroPointEnergy) [reply dev-python/logbook/logbook-1.5.3.ebuild:30]: That is a good question. And I will try to find the answer :-)

**Comment by mgorny** [comment]
So far managed to reproduce the first problem with brotli. I see both packages are installing the same module, so I'll need to add a blocker. Sadly, this means logbook will not be installable on most systems due to brotlipy being required by the http2 stuff.

**Comment by mgorny** [comment]
`dev-python/contextvars` seems to be the culprit. I'll reproduce with upstream tox and report it upstream.

**Comment by ZeroPointEnergy** [comment]
Well the dependencies are optional, so it should not really need brotli anyway unless tests are enabled, which probably isn't the case for most users

**Comment by mgorny** [comment]
Sure, unless the using package needs brotli. However, contextvars seems to cause it to stop working completely. My suggestion would be to add hard blocker on it.

**Comment by ZeroPointEnergy** [comment]
Added the hard blocker and removed the redis-py dependency from the tests

**Comment by ZeroPointEnergy** [comment]
Should I also take over the maintenance as a proxy maintainer? It seems not a lot of people are interested in this package other than me

**Comment by mgorny** [comment]
That is not a hard blocker, and this is not purely a build-time problem.

**Comment by ZeroPointEnergy** [comment]
Ah ok, was not aware there are different kind of blockers. I hope I got the right one now.

**Comment by mgorny** [comment]
> Should I also take over the maintenance as a proxy maintainer? It seems not a lot of people are interested in this package other than me

Please do.

File: dev-python/logbook/logbook-1.5.3.ebuild [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+
+BDEPEND="
+	test? (
+		app-arch/brotli[${PYTHON_USEDEP},python]
+		dev-python/execnet[${PYTHON_USEDEP}]
+		dev-python/jinja[${PYTHON_USEDEP}]
+		dev-python/pyzmq[${PYTHON_USEDEP}]
+		dev-python/sqlalchemy[${PYTHON_USEDEP}]
+	)"
+RDEPEND="!!dev-python/contextvars"
```
Comment (mgorny) [inline dev-python/logbook/logbook-1.5.3.ebuild:32]: The blocker needs to be test-time too, obviously. `d_e_t` would take care of adding RDEPEND to test-depends *if* you called it *after* setting RDEPEND.

File: dev-python/logbook/logbook-1.5.3.ebuild [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+
+LICENSE="BSD"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE=""
+
+distutils_enable_tests pytest
+distutils_enable_sphinx docs
+
+BDEPEND="
```
Comment (mgorny) [inline dev-python/logbook/logbook-1.5.3.ebuild:24]: You're overwriting deps from `distutils_enable*`.

**Comment by mgorny** [comment]
Still one test failing. Back to the list :-).

**Comment by mgorny** [comment]
https://github.com/getlogbook/logbook/issues/310

It's gevent this time.

**Comment by ZeroPointEnergy** [comment]
Added the blocker. Is it normal that a python package causes that much issues with tests? I hope not :-D

**Comment by mgorny** [comment]
Normal, no, but not unknown either.

**Comment by ZeroPointEnergy** [comment]
@mgorny thanks a lot for your patient reviews and help with this :smiley:

**Pull request closed without merging**